subsys/net/ip/net_context.c

```diff
@@ -82,6 +82,32 @@ static enum net_verdict packet_received(struct net_conn *conn,
 static void set_appdata_values(struct net_pkt *pkt, enum net_ip_protocol proto);
 
 #if defined(CONFIG_NET_TCP)
+static int send_reset(struct net_context *context, struct sockaddr *remote);
+
+static struct tcp_backlog_t {
+	struct net_tcp *tcp;
+	struct sockaddr remote;
+	u32_t recv_max_ack;
+	u32_t send_seq;
+	u32_t send_ack;
+	bool cancelled;
+	struct k_delayed_work ack_timer;
+} tcp_backlog[CONFIG_NET_TCP_BACKLOG_SIZE];
+
+static void backlog_ack_timeout(struct k_work *work)
+{
+	struct tcp_backlog_t *tcp_backlog =
```

Comment:
I'm usually ok with variable shadowing, but in this case, I'd find it really confusing. The global array is called tcp_backlog, and pointer to an entry is called the same way. Someone can be well mixed-up. I'd suggest to rename this local var as "backlog_entry" or something.